### PR TITLE
Add help field to variable definitions for enhanced user guidance

### PR DESCRIPTION
**Pull request by tacogips**
## Summary

This PR enhances the template variable system by adding an optional `help` field to variable definitions, providing users with more detailed explanations during interactive prompts. The `description` field continues to be shown in the prompt message by default, while the new `help` field is displayed when users type `?` for additional context.

## Changes

- Added optional `help` field to `VarDef` struct in `internal/template/model/ignconfig.go`
- Updated prompt display logic in `internal/cli/prompt.go` to:
  - Show `description` in the default prompt message for all variable types (string, int, bool)
  - Use `help` field for the `?` help display, falling back to `description` if `help` is not provided
- Updated configuration reference documentation in `docs/reference/configuration.md` to:
  - Document the new `help` field and its purpose
  - Clarify the distinction between `description` (always shown) and `help` (shown on `?`)
  - Add examples showing both fields in use
  - Update TypeScript schema definition

## Changed Files

| File | Additions | Deletions | Change Summary |
| ---- | --------- | --------- | -------------- |
| docs/reference/configuration.md | 11 | 2 | Document help field usage |
| internal/cli/prompt.go | 21 | 5 | Display description in prompts |
| internal/template/model/ignconfig.go | 3 | 1 | Add help field definition |

### Prompt Display Behavior

The prompt system now supports two levels of documentation:

1. **Description** - Always displayed in the prompt message (e.g., `? VARIABLE_NAME - description (required)`)
2. **Help** - Shown when user types `?` during input (uses `help` field if provided, otherwise falls back to `description`)
3. **Example** - Appended to help text if provided

This two-tier approach allows template authors to provide concise inline guidance while offering detailed explanations on demand.

### Implementation Changes

- `VarDef.Help` field is optional (`json:"help,omitempty"`)
- All three prompt types (`promptString`, `promptInt`, `promptBool`) now include description in the message
- The help text resolution follows: `help` field → `description` field → empty
- Example values are appended to the help text regardless of which field is used

### Documentation Updates

- Updated variable definition table to include the new `help` field
- Added "Prompt Display Behavior" section explaining the relationship between `description` and `help`
- Updated TypeScript schema with proper typing for `help` field
- Added concrete examples showing both fields in realistic scenarios (project name, Go module path)

## Related Issues/PRs

---

## Additional Notes

Add your additional notes here

### CODE REVIEW

File: internal/template/model/types.go [hunk 1]
```diff
@@ -10,6 +10,8 @@ const (
 	VarTypeString VarType = "string"
 	// VarTypeInt represents an integer variable type.
 	VarTypeInt VarType = "int"
+	// VarTypeNumber represents a floating-point number variable type (parsed as float64).
+	VarTypeNumber VarType = "number"
```
Comment (tacogips) [inline internal/template/model/types.go]: ## [REVIEW] Medium: Missing test case for VarTypeNumber constant

**Severity**: Medium

### Problem
The newly added `VarTypeNumber` constant in `types.go` is not tested in the `TestVarType_Constants` test function located in `internal/template/model/model_test.go`.

### Why this is a problem
- The test `TestVarType_Constants` (lines 10-28 in model_test.go) validates that VarType constants have the correct string representation
- It currently tests VarTypeString, VarTypeInt, and VarTypeBool, but excludes VarTypeNumber
- This creates incomplete test coverage for the new type constant
- Future refactoring might accidentally break the VarTypeNumber constant without detection
- Inconsistent with testing patterns for other VarType constants

### Problematic code pattern
```go
// In internal/template/model/model_test.go:10-28
func TestVarType_Constants(t *testing.T) {
    tests := []struct {
        name     string
        varType  VarType
        expected string
    }{
        {"string type", VarTypeString, "string"},
        {"int type", VarTypeInt, "int"},
        {"bool type", VarTypeBool, "bool"},
        // MISSING: {"number type", VarTypeNumber, "number"},
    }
    // ...
}
```

### Suggested direction
- Add a test case for VarTypeNumber in the TestVarType_Constants test
- The test case should verify that `VarTypeNumber` has the string value "number"
- This ensures consistency with the other VarType constant tests

File: docs/reference/template-syntax.md [hunk 2]
```diff
@@ -77,11 +77,12 @@ Variable with explicit type validation. The variable must be defined and must ma
 @ign-var:VARIABLE_NAME:TYPE@
 ```
 
-**Supported Types:** `string`, `int`, `bool`
+**Supported Types:** `string`, `int`, `number`, `bool`
 
 **Example Template:**
 ```go
 const Port = @ign-var:port:int@
+const RateLimit = @ign-var:rate_limit:number@
```
Comment (tacogips) [inline docs/reference/template-syntax.md]: ## [REVIEW] Medium: Missing number type example in template-syntax.md

**Severity**: Medium

### Problem
While the documentation adds `number` to the supported types list, the examples section only shows `int`, `bool`, and `string` usage patterns. There is no example demonstrating how to use the `number` type with floating-point values.

### Why this is a problem
- Users may not understand the difference between `int` and `number` types
- The single example of `rate_limit` is not enough to demonstrate typical usage patterns
- Missing validation examples for `min_float` and `max_float` constraints

### Problematic code pattern
```go
const Port = @ign-var:port:int@
const RateLimit = @ign-var:rate_limit:number@  // Only example added
const Debug = @ign-var:debug:bool@
const Name = "@ign-var:name:string@"
```

### Suggested direction
Add a comprehensive example showing:
- Number type usage with floating-point default values
- Validation with min_float/max_float constraints
- Explanation of when to use `int` vs `number`
- Example in ign.json showing number variable definition with constraints

File: internal/cli/prompt.go [hunk 3]
```diff
@@ -186,10 +198,97 @@ func promptInt(name string, varDef model.VarDef, help string) (int, error) {
 	return strconv.Atoi(result)
 }
 
+// promptNumber prompts for a floating-point number variable.
```
Comment (tacogips) [inline internal/cli/prompt.go]: ## [REVIEW] High: Missing test coverage for promptNumber function

**Severity**: High

### Problem
The newly added promptNumber function (lines 201-280) has no corresponding unit tests.

### Why this is a problem
- The function contains complex logic including float parsing, validation, min/max constraint checking, and type conversion from multiple numeric types (float64, float32, int)
- Without tests, it is impossible to verify the function behaves correctly for edge cases (empty input, invalid input, boundary values, type conversions)
- Other similar functions in the codebase (promptInt, promptString) should have test coverage for consistency
- Validation logic bugs could allow invalid values to be accepted or valid values to be rejected

### Problematic code pattern
```go
func promptNumber(name string, varDef model.VarDef, help string) (float64, error) {
    // 80 lines of logic without test coverage
    // Type conversion, validation, parsing, etc.
}
```

### Suggested direction
- Create internal/cli/prompt_test.go with test cases for promptNumber
- Test cases should cover: valid float input, empty input with/without required flag, invalid input, min/max boundary validation, type conversion from int/float32/float64 defaults

File: internal/config/validation.go [hunk 4]
```diff
@@ -283,7 +303,30 @@ func validateVariables(variables map[string]model.VarDef) error {
 			}
 		}
 
-		// Validate example value against min/max constraints
+		// Validate default value against min_float/max_float constraints for numbers
+		if varDef.Default != nil && varDef.Type == model.VarTypeNumber {
+			floatVal := toFloat64(varDef.Default)
+			if floatVal != nil {
+				if varDef.MinFloat != nil && *floatVal < *varDef.MinFloat {
```
Comment (tacogips) [inline internal/config/validation.go]: ## [REVIEW] Medium: Silent failure when toFloat64 returns nil

**Severity**: Medium

### Problem
The validation logic silently skips validation when `toFloat64()` returns nil (when the value cannot be converted to float64). This happens in both the default value validation (line 308-327) and example value validation (line 353-372).

### Why this is a problem
- If `varDef.Default` or `varDef.Example` contains a value that cannot be converted to float64, the min_float/max_float constraints are silently ignored
- This could occur if the JSON unmarshaling produces an unexpected type that passes the `validateValueType()` check but fails `toFloat64()`
- The silent failure provides no feedback to users about invalid configuration
- This creates an inconsistent validation state where a value passes validation but doesn't actually satisfy the constraints

### Problematic code pattern
```go
if varDef.Default != nil && varDef.Type == model.VarTypeNumber {
    floatVal := toFloat64(varDef.Default)
    if floatVal != nil {  // Silent skip when nil
        // validation happens
    }
    // No error when floatVal is nil - validation silently skipped
}
```

### Suggested direction
Consider one of these approaches:
1. Return an error when `toFloat64()` returns nil for a number type variable (strict validation)
2. Add a comment explaining why silent skip is acceptable (if there's a valid reason)
3. Log a warning when constraints are skipped due to type conversion failure

The first approach is recommended for consistency with the project's strict validation philosophy.

File: docs/reference/template-syntax.md [hunk 5]
```diff
@@ -77,11 +77,12 @@ Variable with explicit type validation. The variable must be defined and must ma
 @ign-var:VARIABLE_NAME:TYPE@
 ```
 
-**Supported Types:** `string`, `int`, `bool`
+**Supported Types:** `string`, `int`, `number`, `bool`
```
Comment (tacogips) [inline docs/reference/template-syntax.md]: ## [REVIEW] Medium: Incomplete help field behavior documentation

**Severity**: Medium

### Problem
The help field is introduced with behavior description in configuration.md, but there is no corresponding documentation in template-syntax.md about how users interact with the help text during template generation.

### Why this is a problem
- Users reading template-syntax.md will not know about the help field feature
- The interactive behavior (typing `?` to see help) is only documented in configuration.md
- Template authors need to understand when to use `description` vs `help` fields

### Current documentation gap
configuration.md lines 122-124 explain the behavior:
```markdown
- The `description` is always displayed in the prompt message
- When user types `?`, the `help` text is shown
- If `example` is provided, it is appended to the help text
```

But template-syntax.md has no mention of this interactive feature.

### Suggested direction
Add a section in template-syntax.md explaining:
- The purpose of the help field in variable definitions
- How users access help text during template generation
- Best practices for writing concise descriptions vs detailed help text
- Examples showing good description/help field usage

File: internal/config/validation_test.go [hunk 6]
```diff
@@ -257,6 +257,77 @@ func TestValidateVariables(t *testing.T) {
...
+				Type:        model.VarTypeNumber,
+				Description: "Rate",
+				Default:     15.0,
+				MaxFloat:    floatPtr(10.0),
+			},
+		}
+		if err := validateVariables(vars); err == nil {
+			t.Error("Expected error for default value above max_float")
+		}
+	})
```
Comment (tacogips) [inline internal/config/validation_test.go]: ## [REVIEW] High: Missing test coverage for Example value validation against MinFloat/MaxFloat

**Severity**: High

### Problem
The new test cases for `VarTypeNumber` do not include tests for validating `Example` values against `MinFloat`/`MaxFloat` constraints.

### Why this is a problem
- The validation logic in `internal/config/validation.go` (lines 351-372) validates Example values against MinFloat/MaxFloat constraints
- Without test coverage, this validation path is untested and could break without detection
- This creates an inconsistency: Default values are tested (lines 304-330), but Example values are not
- Violates the code review guideline: "Check if tests have been created to assert changes for newly added or modified functions"

### Problematic code pattern
```go
// Tests for Default value constraints exist (lines 304-330):
t.Run("number default below min_float", func(t *testing.T) { ... })
t.Run("number default above max_float", func(t *testing.T) { ... })

// But NO tests for Example value constraints
// Missing: t.Run("number example below min_float", ...)
// Missing: t.Run("number example above max_float", ...)
```

### Suggested direction
- Add test cases similar to "number default below min_float" and "number default above max_float"
- Name them "number example below min_float" and "number example above max_float"
- Use Example field instead of Default field
- Verify that validation errors are returned when Example violates MinFloat/MaxFloat constraints

File: internal/cli/prompt.go [hunk 7]
```diff
@@ -186,10 +198,97 @@ func promptInt(name string, varDef model.VarDef, help string) (int, error) {
...
+		}
+
+		return nil
+	}
+
+	if err := survey.AskOne(prompt, &result, survey.WithValidator(numberValidator)); err != nil {
+		return 0, err
+	}
+
+	if result == "" {
```
Comment (tacogips) [inline internal/cli/prompt.go]: ## [REVIEW] Medium: Ambiguous zero return for empty non-required number input

**Severity**: Medium

### Problem
When a number variable is not required and the user provides empty input, the function returns (0, nil). This makes it impossible to distinguish between a user explicitly entering 0 versus leaving the field empty.

### Why this is a problem
- Template processing cannot differentiate between "user provided 0" and "user provided nothing"
- This may cause issues if template logic needs to check whether a value was actually provided
- The same issue exists in promptInt (lines 194-196) but inheriting the problem does not make it correct
- In Go, this is typically handled by returning a pointer type (nil for not provided, non-nil for provided)

### Problematic code pattern
```go
if result == "" {
    return 0, nil  // Ambiguous: was it 0 or empty?
}
return strconv.ParseFloat(result, 64)
```

### Suggested direction
- Consider whether the system needs to distinguish "not provided" from "zero value"
- If yes, change signature to return (*float64, error) or use a struct with a "provided" flag
- If no, add a comment explaining that empty input is treated as zero for non-required fields
- Ensure this behavior is documented and consistent across all prompt functions

File: docs/reference/template-syntax.md (hunk 5 above)
Comment (tacogips) [inline docs/reference/template-syntax.md]: ## [REVIEW] High: Variable Types table missing number type entry

**Severity**: High

### Problem
The Variable Types reference table in Appendix B (line 1111-1117) is missing an entry for the `number` type that was added in this PR.

### Why this is a problem
- The table is incomplete and will mislead users
- Users looking at the quick reference will not know about the number type
- The table contradicts the earlier documentation that lists number as a supported type

### Problematic code pattern
```markdown
| Type | JSON Type | Example | Notes |
|------|-----------|---------|-------|
| `string` | `string` | `"hello"` | Any text, preserved as-is |
| `int` | `number` | `8080` | Integer values only |
| `bool` | `boolean` | `true`, `false` | Used in conditionals |
```

Missing: `number` type entry

### Suggested direction
Add a row for the `number` type:
```markdown
| `number` | `number` | `1.5`, `3.14` | Floating-point values, validated with min_float/max_float |
```

Ensure the notes clearly distinguish between `int` and `number` types.

File: internal/cli/prompt.go [hunk 8]
```diff
@@ -186,10 +198,97 @@ func promptInt(name string, varDef model.VarDef, help string) (int, error) {
...
+
+	return strconv.ParseFloat(result, 64)
+}
+
 // promptBool prompts for a boolean variable.
 func promptBool(name string, varDef model.VarDef, help string) (bool, error) {
 	var result bool
 
+	// Build message with description displayed by default
+	message := name
```
Comment (tacogips) [inline internal/cli/prompt.go]: ## [REVIEW] Low: Missing required indicator in promptBool function

**Severity**: Low

### Problem
The promptBool function does not display a "(required)" indicator in the prompt message, unlike promptString, promptInt, and promptNumber functions.

### Why this is a problem
- Inconsistent user experience across different variable types
- Users cannot easily identify whether a boolean variable is required or optional
- Other prompt functions (promptString, promptInt, promptNumber) all show "(required)" when varDef.Required is true
- This inconsistency may confuse template users

### Problematic code pattern
```go
// promptBool (lines 287-290)
message := name
if varDef.Description != "" {
    message += " - " + varDef.Description
}
// Missing: if varDef.Required { message += " (required)" }

// Compare to promptString (lines 79-85)
message := name
if varDef.Description != "" {
    message += " - " + varDef.Description
}
if varDef.Required {
    message += " (required)"  // <-- Has required indicator
}
```

### Suggested direction
- Add the same "(required)" indicator logic to promptBool for consistency
- Ensure all prompt functions have consistent message formatting

File: internal/config/validation_test.go [hunk 9]
```diff
@@ -257,6 +257,77 @@ func TestValidateVariables(t *testing.T) {
...
+				Type:        model.VarTypeNumber,
+				Description: "Rate limit per second",
+				Default:     1.5,
+				MinFloat:    floatPtr(0.1),
+				MaxFloat:    floatPtr(100.0),
+			},
+		}
+		if err := validateVariables(vars); err != nil {
+			t.Errorf("Valid number variable should pass validation: %v", err)
+		}
```
Comment (tacogips) [inline internal/config/validation_test.go]: ## High: Cross-File Issue: Incomplete Test Coverage for toFloat64 Type Coercion Edge Cases

**Severity**: High
**Affected Files**:
- `internal/config/validation.go`
- `internal/config/validation_test.go`

### Problem
The `toFloat64` helper function (lines 419-439 in validation.go) handles multiple numeric types (float64, float32, int, int32, int64), but there are NO dedicated tests verifying this type coercion works correctly across all supported types when validating number variables with min_float/max_float constraints.

### Why this is a problem
- Type coercion is a critical path that affects validation correctness
- JSON unmarshaling can produce different numeric types (int, float64, etc.)
- Without explicit tests, we cannot verify that an integer value (e.g., 5) correctly validates against float constraints (e.g., min_float: 1.0, max_float: 10.0)
- The existing tests only use float64 literals (e.g., 0.5, 15.0, 1.5) but don't test integer-as-number cases

### Code comparison

**Implementation**: `internal/config/validation.go` (lines 419-439)
```go
// toFloat64 converts a value to float64 pointer if it's a numeric type.
func toFloat64(value interface{}) *float64 {
    switch v := value.(type) {
    case float64:
        return &v
    case float32:
        f := float64(v)
        return &f
    case int:       // Used for JSON numbers without decimals
        f := float64(v)
        return &f
    // ... also handles int32, int64
    default:
        return nil  // Non-numeric types return nil
    }
}
```

**Test File**: `internal/config/validation_test.go` (lines 304-330)
```go
t.Run("number default below min_float", func(t *testing.T) {
    vars := map[string]model.VarDef{
        "rate": {
            Type:        model.VarTypeNumber,
            Description: "Rate",
            Default:     0.5,  // Only tests float64 literals
            MinFloat:    floatPtr(1.0),
        },
    }
    // Missing: Test with Default: 5 (int) against float constraints
})
```

### Suggested direction
- Add test case: "number variable with int default value against float constraints" (e.g., Default: 5, MinFloat: 1.0, MaxFloat: 10.0)
- Add test case: "number variable with int default below min_float" (e.g., Default: 0, MinFloat: 1.0)
- Add test case: "number variable with int default above max_float" (e.g., Default: 100, MaxFloat: 10.0)
- Verify toFloat64 handles nil pointer case (when value is non-numeric)

Comment (tacogips) [inline internal/config/validation_test.go]: ## High: Cross-File Issue: Incomplete Test Coverage for toFloat64 Type Coercion Edge Cases

**Severity**: High
**Affected Files**:
- `internal/config/validation.go`
- `internal/config/validation_test.go`

### Problem
The `toFloat64` helper function (lines 419-439 in validation.go) handles multiple numeric types (float64, float32, int, int32, int64), but there are NO dedicated tests verifying this type coercion works correctly across all supported types when validating number variables with min_float/max_float constraints.

### Why this is a problem
- Type coercion is a critical path that affects validation correctness
- JSON unmarshaling can produce different numeric types (int, float64, etc.)
- Without explicit tests, we cannot verify that an integer value (e.g., 5) correctly validates against float constraints (e.g., min_float: 1.0, max_float: 10.0)
- The existing tests only use float64 literals (e.g., 0.5, 15.0, 1.5) but do not test integer-as-number cases

### Code comparison

**Implementation**: `internal/config/validation.go` (lines 419-439)
```go
// toFloat64 converts a value to float64 pointer if it's a numeric type.
func toFloat64(value interface{}) *float64 {
    switch v := value.(type) {
    case float64:
        return &v
    case float32:
        f := float64(v)
        return &f
    case int:       // Used for JSON numbers without decimals
        f := float64(v)
        return &f
    // ... also handles int32, int64
    default:
        return nil  // Non-numeric types return nil
    }
}
```

**Test File**: `internal/config/validation_test.go` (lines 304-330)
```go
t.Run("number default below min_float", func(t *testing.T) {
    vars := map[string]model.VarDef{
        "rate": {
            Type:        model.VarTypeNumber,
            Description: "Rate",
            Default:     0.5,  // Only tests float64 literals
            MinFloat:    floatPtr(1.0),
        },
    }
    // Missing: Test with Default: 5 (int) against float constraints
})
```

### Suggested direction
- Add test case: "number variable with int default value against float constraints" (e.g., Default: 5, MinFloat: 1.0, MaxFloat: 10.0)
- Add test case: "number variable with int default below min_float" (e.g., Default: 0, MinFloat: 1.0)
- Add test case: "number variable with int default above max_float" (e.g., Default: 100, MaxFloat: 10.0)
- Verify toFloat64 handles nil pointer case (when value is non-numeric)

File: docs/reference/template-syntax.md (hunk 5 above)
Comment (tacogips) [inline docs/reference/template-syntax.md]: ## High: Variable Types reference table missing number type

**Severity**: High
**Affected Files**:
- `docs/reference/template-syntax.md` (this line 80, and table at line ~172)
- `docs/reference/configuration.md` (line 109)
- `internal/template/model/types.go` (line 13-14)

### Problem
This line correctly adds `number` to the supported types list, but the detailed **Variable Types reference table** in section 1.6 (around line 172) was NOT updated and still only shows 3 types (string, int, bool).

### Why this is a problem
- This line now states: **Supported Types:** `string`, `int`, `number`, `bool`
- The Variable Types table further down only has rows for: string, int, bool (missing number)
- Users reading the reference table will not discover the number type exists
- Internal documentation inconsistency within the same file

### Code comparison

**This line 80** - CORRECT:
```markdown
**Supported Types:** `string`, `int`, `number`, `bool`
```

**Section 1.6 table (line ~172)** - INCOMPLETE:
```markdown
| Type | Example Value | JSON Type | Usage |
| string | "hello" | string | Text values |
| int | 8080 | number | Ports, counts |
| bool | true, false | boolean | Feature flags |
(MISSING: number type row)
```

**types.go** confirms number type exists:
```go
VarTypeNumber VarType = "number"
```

### Suggested direction
Add a row to the Variable Types table in section 1.6 showing the number type with example value 1.5, JSON type number, and usage for floating-point values, rates, ratios.

File: internal/cli/prompt.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tacogips) [reply internal/cli/prompt.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Created internal/cli/prompt_test.go with comprehensive test suite for promptNumber function
- Tests cover: valid float input, empty input with/without required flag, invalid input, min/max boundary validation, type conversion from int/float32/float64 defaults
- 306 lines of test coverage added for prompt validation edge cases

Resolving this thread.

File: internal/template/model/types.go (hunk 1 above)
Comment (tacogips) [reply internal/template/model/types.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added test case for VarTypeNumber constant in TestVarType_Constants test
- Test verifies that VarTypeNumber has the string value "number"
- Consistent with other VarType constant tests

Resolving this thread.

File: internal/config/validation.go (hunk 4 above)
Comment (tacogips) [reply internal/config/validation.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Changed validation logic to return error when toFloat64 returns nil for number type variables
- Prevents silent constraint skipping by enforcing strict validation
- Added comment explaining the error is returned to prevent silent failure
- Error message: "default/example value cannot be converted to number type"

Resolving this thread.

File: docs/reference/template-syntax.md (hunk 5 above)
Comment (tacogips) [reply docs/reference/template-syntax.md]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added comprehensive help field documentation in docs/reference/template-syntax.md
- Section 1.6 explains: purpose, interactive behavior (typing '?'), best practices
- Examples show good description/help field usage with project_name and go_module_path
- Documents how description is always shown, help shown on demand

Resolving this thread.

Comment (tacogips) [reply docs/reference/template-syntax.md]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Updated Variable Types table in docs/reference/template-syntax.md (Section 1.7)
- Added number type entry: `number` | `1.5`, `3.14` | `number` | Floating-point values, rates, percentages, decimals
- Table now includes all four types: string, int, number, bool
- Notes clearly distinguish between int (integers only) and number (floating-point)

Resolving this thread.

File: internal/cli/prompt.go (hunk 8 above)
Comment (tacogips) [reply internal/cli/prompt.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added required indicator to promptBool function in internal/cli/prompt.go
- Now displays "(required)" when varDef.Required is true
- Consistent with promptString, promptInt, and promptNumber functions
- Uniform user experience across all variable types

Resolving this thread.

File: internal/config/validation_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (tacogips) [reply internal/config/validation_test.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added comprehensive test coverage for number example value validation in internal/config/validation_test.go
- Tests include: number example below min_float (line 333-343), number example above max_float (line 347-357)
- Tests mirror the pattern used for default value validation
- Verifies validation errors are returned when example violates MinFloat/MaxFloat constraints
- Total of 88 new test lines added for number validation edge cases

Resolving this thread.

File: docs/implementation/architecture.md [hunk 10]
```diff
@@ -493,6 +493,7 @@ type VarType string
 const (
     VarTypeString VarType = "string"
     VarTypeInt    VarType = "int"
+    VarTypeNumber VarType = "number"
```
[the remark this answers is not in the document]
Comment (tacogips) [reply docs/implementation/architecture.md]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added Help, MinFloat, and MaxFloat fields to VarDef struct documentation in architecture.md
- Updated struct definition to match configuration.md changes

Resolving this thread.

File: internal/cli/prompt.go (hunk 3 above)
Comment (tacogips) [reply internal/cli/prompt.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Created internal/cli/prompt_test.go with comprehensive test coverage for promptNumber function
- Same fix as the other promptNumber test coverage comment

Resolving this thread.

File: internal/config/validation.go (hunk 4 above)
Comment (tacogips) [reply internal/config/validation.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Changed validation to return error when toFloat64 returns nil
- Prevents silent constraint skipping for number type validation
- Added explicit error handling for type conversion failures

Resolving this thread.

File: docs/reference/template-syntax.md (hunk 5 above)
Comment (tacogips) [reply docs/reference/template-syntax.md]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added Section 1.6 in template-syntax.md documenting help field purpose
- Documented interactive behavior (typing ? to see help)
- Added best practices for writing description vs help text

Resolving this thread.

File: internal/config/validation_test.go (hunk 6 above)
Comment (tacogips) [reply internal/config/validation_test.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added test cases for Example value validation against MinFloat/MaxFloat constraints
- Tests for example below min_float and example above max_float now included

Resolving this thread.

File: docs/reference/template-syntax.md (hunk 5 above)
Comment (tacogips) [reply docs/reference/template-syntax.md]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Updated Variable Types table in Appendix B to include number type
- Added example values (1.5, 3.14) and usage description

Resolving this thread.

File: docs/reference/configuration.md [hunk 11]
```diff
@@ -86,6 +86,12 @@ Defines template metadata, required variables, and template-specific settings.
       "default": 8080,
       "required": false
     },
+    "rate_limit": {
+      "type": "number",
+      "description": "Rate limit per second",
+      "default": 1.5,
+      "required": false
+    },
```
[the remark this answers is not in the document]
Comment (tacogips) [reply docs/reference/configuration.md]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Enhanced rate_limit example to include min_float (0.1) and max_float (100.0) constraints
- Demonstrates proper usage of number type with validation

Resolving this thread.

File: internal/cli/prompt.go (hunk 8 above)
Comment (tacogips) [reply internal/cli/prompt.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added required indicator to promptBool function for consistency with other prompt functions

Resolving this thread.

File: internal/config/validation_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (tacogips) [reply internal/config/validation_test.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added comprehensive test coverage for number example values against min_float/max_float constraints
- Tests include example below min_float and example above max_float

Resolving this thread.

File: internal/config/validation_test.go (hunk 9 above)
Comment (tacogips) [reply internal/config/validation_test.go]: Fixed in commit 3dd5973 (PR #15)

Changes made:
- Added test coverage for toFloat64 type coercion edge cases
- Tests include int default values below/above/within min_float/max_float ranges

Resolving this thread.

File: docs/reference/configuration.md [hunk 12]
```diff
@@ -100,14 +106,22 @@ Defines template metadata, required variables, and template-specific settings.
 
 | Field | Type | Required | Description |
 |-------|------|----------|-------------|
-| `type` | string | Yes | Variable type: `string`, `int`, or `bool` |
-| `description` | string | Yes | Human-readable description |
+| `type` | string | Yes | Variable type: `string`, `int`, `number`, or `bool` |
```
[the remark this answers is not in the document]
Comment (tacogips) [reply docs/reference/configuration.md]: Fixed in commit d2a573f (PR #16)

Changes made:
- Changed type field format from backticks for each type individually (`string`, `int`, `number`, `bool`) to union type format (`string` | `int` | `number` | `bool`) for consistency with TypeScript interface style

Resolving this thread.

File: internal/config/validation_test.go (hunk 9 above)
Comment (tacogips) [reply internal/config/validation_test.go]: Fixed in commit d2a573f (PR #16)

Changes made:
- Added test case 'number variable with int default below min' testing Default: 0 against Min: 1.0
- Added test case 'number variable with int default above max' testing Default: 100 against Max: 10.0
- Added test case 'number variable with int default in valid range' testing Default: 5 against Min: 1.0, Max: 10.0

Resolving this thread.

File: internal/config/validation_test.go [hunk 13]
```diff
@@ -257,6 +257,77 @@ func TestValidateVariables(t *testing.T) {
...
 	})
+
+	t.Run("valid number variable", func(t *testing.T) {
+		vars := map[string]model.VarDef{
+			"rate_limit": {
+				Type:        model.VarTypeNumber,
+				Description: "Rate limit per second",
+				Default:     1.5,
+				MinFloat:    floatPtr(0.1),
+				MaxFloat:    floatPtr(100.0),
```
[the remark this answers is not in the document]
Comment (tacogips) [reply internal/config/validation_test.go]: Fixed in commit d2a573f (PR #16)

Changes made:
- Added test case 'number variable with both default and example in valid range' with Default: 1.5, Example: 2.5, Min: 0.1, Max: 100.0

Resolving this thread.

File: internal/config/validation_test.go [hunk 14]
```diff
@@ -257,6 +257,77 @@ func TestValidateVariables(t *testing.T) {
 			t.Error("Expected error for min > max")
 		}
 	})
+
```
[the remark this answers is not in the document]
Comment (tacogips) [reply internal/config/validation_test.go]: Fixed in commit d2a573f (PR #16)

Changes made:
- Added test case 'number example below min' testing Example: 0.5 against Min: 1.0
- Added test case 'number example above max' testing Example: 15.0 against Max: 10.0
- Achieved complete test symmetry with integer validation test patterns

Resolving this thread.

File: internal/template/model/ignconfig.go [hunk 15]
```diff
@@ -24,10 +24,12 @@ type IgnJson struct {
...
 // VarDef defines a template variable with validation rules.
 type VarDef struct {
-	// Type is the variable type (string, int, or bool).
+	// Type is the variable type (string, int, number, or bool).
 	Type VarType `json:"type"`
-	// Description is a human-readable description of the variable.
+	// Description is a short description shown in the prompt by default.
 	Description string `json:"description"`
+	// Help is a longer explanation shown when user types '?' during input.
+	Help string `json:"help,omitempty"`
```
[the remark this answers is not in the document]
Comment (tacogips) [reply internal/template/model/ignconfig.go]: Fixed in commit d2a573f (PR #16)

Changes made:
- Added Help field to VarDef struct in architecture.md section 5.1
- Updated Min and Max fields from *int to *float64 to support both int and number types
- Added comments indicating Min/Max apply to both int and number variable types

Resolving this thread.

File: docs/reference/configuration.md [hunk 16]
```diff
@@ -100,14 +106,22 @@ Defines template metadata, required variables, and template-specific settings.
```
[the remark this answers is not in the document]
Comment (tacogips) [reply docs/reference/configuration.md]: Fixed in commit d2a573f (PR #16)

Changes made:
- Added documentation to 'Prompt Display Behavior' section explaining range display for int variables: [min-max], [>=min], or [<=max]
- Added documentation for number (floating-point) variable range display using same format with decimal values: [0.0-10.0], [>=1.0], or [<=100.0]
- Added note that validation errors show the constraint when input is out of range

Resolving this thread.
